source/opt/common_uniform_elim_pass.cpp

```diff
@@ -356,7 +356,20 @@ bool CommonUniformElimPass::CommonUniformLoadElimination(ir::Function* func) {
     // new load insertion point. Trying to keep register pressure down.
     if (mergeBlockId == bp->id()) {
       mergeBlockId = 0;
-      insertItr = bp->begin();
+
+      // Update insertItr as |bp->begin()| only when it will not be removed.
+      // Without this check, it can set |insertItr| as a dangling pointer.
```

Comment:
Could the original value of insertItr from line 351 have the same problem?  If so, where would you put the insertion point?

Reply:
Yes you are right. I updated it.